src/com/lftechnology/java/training/alina/jdbc/dao/employee/impl/EmployeeDaoImpl.java

```diff
@@ -0,0 +1,162 @@
+            preparedStatement.setString(2, employee.getDepartment());
+            preparedStatement.setString(3, employee.getAddress());
+            preparedStatement.setString(4, employee.getRole());
+            preparedStatement.setInt(5, employee.getUserId());
+            preparedStatement.setTimestamp(6, employee.getCreatedAt());
+            int result = preparedStatement.executeUpdate();
+            ResultSet rs = preparedStatement.getGeneratedKeys();
+            if (result > 0) {
+                if (rs.next()) {
+                    employee.setEmployeeId(rs.getInt(1));
+                }
+            }
+            preparedStatement.close();
+        } catch (SQLException sqe) {
+            LOGGER.log(Level.WARNING, "SQLException : {0}", new Object[] { sqe });
+        }
+        return employee;
+    }
+
+    @Override
+    public Integer update(Database database) {
+        int resultSet = 0;
+        try {
+            Connection connection = DbFacade.getDbConnection();
+            PreparedStatement preparedStatement = connection.prepareStatement(database.getSqlQuery());
+            preparedStatement = DbFacade.setParameterizedObjects(database, preparedStatement);
+            resultSet = preparedStatement.executeUpdate();
+            preparedStatement.close();
+        } catch (SQLException sqe) {
+            LOGGER.log(Level.WARNING, "SQLException : {0}", new Object[] { sqe });
+        }
+        return resultSet;
+    }
+
+    @Override
+    public Boolean delete(String username) {
+        boolean isDeleted = false;
+        String sql = "update employee set is_deleted=?,modified_at=? where (fullname=? and is_deleted=?)";
+        try {
+            Connection connection = DbFacade.getDbConnection();
+            PreparedStatement preparedStatement = connection.prepareStatement(sql);
+            preparedStatement.setBoolean(1, Constants.DELETED);
+            preparedStatement.setTimestamp(2, DateTimeService.getCurrentTimeStamp());
+            preparedStatement.setString(3, username);
+            preparedStatement.setBoolean(4, Constants.NOT_DELETED);
+            int result = preparedStatement.executeUpdate();
+            isDeleted = (result != 0) ? true : false;
+            preparedStatement.close();
+        } catch (SQLException sqe) {
+            LOGGER.log(Level.WARNING, "SQLException : {0}", new Object[] { sqe });
+        }
+        return isDeleted;
+    }
+
+    @Override
+    public List<Employee> searchEmployee(String... searchContent) {
+        List<Employee> employeeList = new ArrayList<Employee>();
+        String sql =
+                "select * from employee e inner join user u on e.user_id=u.user_id where (e.fullname like ? or e.department like ? or e.address like ?) and (u.is_terminated=? or e.is_deleted=?)";
+        try {
+            Connection connection = DbFacade.getDbConnection();
+            PreparedStatement preparedStatement = (PreparedStatement) connection.prepareStatement(sql);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this unnecessary cast to "PreparedStatement". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1905)